test/sync/src/commonMain/kotlin/io/realm/test/mongodb/TestApp.kt

```diff
@@ -41,20 +45,29 @@ const val TEST_APP_1 = "testapp1" // Id for the default test app
  * @param builder the builder used to build the final app. The builder is already primed with the
  * default test app configuration, but can be used to override the defaults and add additional
  * options.
+ * @param fileSystem platform-dependent entry point to allow file cleanup after the test.
  * @param debug enable trace of command server and rest api calls in the test app.
  */
+@Suppress("LongParameterList")
 class TestApp(
     appName: String = TEST_APP_1,
     dispatcher: CoroutineDispatcher = singleThreadDispatcher("test-app-dispatcher"),
     appId: String = runBlocking(dispatcher) { getAppId(appName, debug) },
     logLevel: LogLevel = LogLevel.WARN,
     builder: (AppConfiguration.Builder) -> AppConfiguration.Builder = { it },
+    private val fileSystem: FileSystem,
     debug: Boolean = false
 ) : App by App.create(builder(testAppConfigurationBuilder(appId, logLevel)).dispatcher(dispatcher).build()),
     AdminApi by (runBlocking(dispatcher) { AdminApiImpl(TEST_SERVER_BASE_URL, appId, debug, dispatcher) }) {
 
     fun close() {
         deleteAllUsers()
+
+        // Make sure to clear cached apps before deleting files
+        RealmInterop.realm_clear_cached_apps()
+
+        // Delete metadata Realm files
+        fileSystem.deleteRecursively((appFilesDirectory() + "/mongodb-realm").toPath())
```

Comment:
This looks a bit coincidental. We don't specify `appFilesDirectory` as part of the AppConfiguration. Are we sure that it always works on all platforms?

Reply:
I added jvmTest (sync didn't have it yet 🙊) and it does delete the files 👍

Reply:
Ok. I guess manually validating it on `iOS` too would be ok for now. 

It would be nice to understand how the this worked, but I guess either this data is only created along with the realm or there must be something corresponding to our `appFilesDirectory` somewhere in core ... maybe just `./` somehow ending up the same place 🤔